pallets/msa/src/lib.rs

```diff
@@ -159,6 +185,33 @@ impl<T: Config> Pallet<T> {
 
 		Ok(())
 	}
+
+	pub fn revoke_key(key: &T::AccountId) -> DispatchResult {
+		KeyInfoOf::<T>::try_mutate(key, |maybe_info| -> DispatchResult {
+			let mut info = maybe_info.take().ok_or(Error::<T>::NoKeyExist)?;
+
+			ensure!(info.expired == T::BlockNumber::zero(), Error::<T>::KeyRevoked);
+
+			let current_block = frame_system::Pallet::<T>::block_number();
+
+			info.expired = current_block;
+
+			*maybe_info = Some(info);
+
+			Ok(())
+		})?;
+
+		Ok(())
+	}
+
+	pub fn try_get_key_info(key: &T::AccountId) -> Result<KeyInfo<T::BlockNumber>, DispatchError> {
+		let info = Self::get_key_info(key).ok_or(Error::<T>::NoKeyExist)?;
+		Ok(info)
+	}
+
+	pub fn get_owner_of(key: &T::AccountId) -> Option<MessageSenderId> {
```

Comment:
This function isn't being used anywhere in this file right now.

Reply:
Its used below. The window needs to be expand.